driver-core/src/main/com/mongodb/internal/connection/CommandMessage.java

```diff
@@ -201,27 +222,49 @@ MongoNamespace getNamespace() {
     }
 
     @Override
-    protected EncodingMetadata encodeMessageBodyWithMetadata(final BsonOutput bsonOutput, final OperationContext operationContext) {
+    protected EncodingMetadata encodeMessageBodyWithMetadata(final ByteBufferBsonOutput bsonOutput, final OperationContext operationContext) {
         int messageStartPosition = bsonOutput.getPosition() - MESSAGE_PROLOGUE_LENGTH;
         int commandStartPosition;
         if (useOpMsg()) {
             int flagPosition = bsonOutput.getPosition();
             bsonOutput.writeInt32(0);   // flag bits
             bsonOutput.writeByte(0);    // payload type
             commandStartPosition = bsonOutput.getPosition();
-
-            addDocument(command, bsonOutput, commandFieldNameValidator, getExtraElements(operationContext));
-
-            if (payload != null) {
-                bsonOutput.writeByte(1);          // payload type
-                int payloadBsonOutputStartPosition = bsonOutput.getPosition();
-                bsonOutput.writeInt32(0);         // size
-                bsonOutput.writeCString(payload.getPayloadName());
-                writePayload(new BsonBinaryWriter(bsonOutput, payloadFieldNameValidator), bsonOutput, getSettings(),
-                        messageStartPosition, payload, getSettings().getMaxDocumentSize());
-
-                int payloadBsonOutputLength = bsonOutput.getPosition() - payloadBsonOutputStartPosition;
-                bsonOutput.writeInt32(payloadBsonOutputStartPosition, payloadBsonOutputLength);
+            ArrayList<BsonElement> extraElements = getExtraElements(operationContext);
+            // `DualMessageSequences` requires validation only if no response is expected, otherwise we must rely on the server validation
+            boolean validateDocumentSizeLimits = !(sequences instanceof DualMessageSequences) || !responseExpected;
```

Comment:
This exception makes the new `bulkWrite` command behavior regarding BSON size validation inconsistent with all other commands. I am trying to figure this out in https://mongodb.slack.com/archives/C035ZJL6CQN/p1730774917892009.

Reply:
This is resolved by https://jira.mongodb.org/browse/JAVA-5695, which is part of the current PR.